gutenberg.php

```diff
@@ -44,30 +196,67 @@ function gutenberg_build_files_notice() {
 }
 
 /**
- * Verify that we can initialize the Gutenberg editor , then load it.
+ * Verify that we can initialize the Gutenberg editor, then load it.
  *
  * @since 1.5.0
  */
 function gutenberg_pre_init() {
-	global $wp_version;
-	if ( defined( 'GUTENBERG_DEVELOPMENT_MODE' ) && GUTENBERG_DEVELOPMENT_MODE && ! file_exists( __DIR__ . '/build/blocks' ) ) {
-		add_action( 'admin_notices', 'gutenberg_build_files_notice' );
-		return;
-	}
-
 	// Get unmodified $wp_version.
 	include ABSPATH . WPINC . '/version.php';
 
-	// Strip '-src' from the version string. Messes up version_compare().
-	$version = str_replace( '-src', '', $wp_version );
+	// Strip '-src' from the WP version string. May interfere with `version_compare()`.
+	$wp_version = str_replace( '-src', '', $wp_version );
+
+	/*
+	 * Note that `version_compare()` considers 1 < 1.0 < 1.0.0 (before suffixes)
+	 * so 6.0 < 6.0.0-alpha and 6.0-beta < 6.0.0-alpha are both true.
+	 * To work around any possible problems when comparing with WP development version strings
+	 * the GUTENBERG_MAX_WP_VERSION is incremented by one and the comparisons are with `>=`.
+	 */
+	$max_supported_version = (float) GUTENBERG_MAX_WP_VERSION + 0.1;
 
-	// Compare against major release versions (X.Y) rather than minor (X.Y.Z)
-	// unless a minor release is the actual minimum requirement. WordPress reports
-	// X.Y for its major releases.
-	if ( version_compare( $version, '5.9', '<' ) ) {
-		add_action( 'admin_notices', 'gutenberg_wordpress_version_notice' );
+	// `version_compare()` expects strings.
+	$max_supported_version = (string) $max_supported_version;
+
+	// Check if this version of Gutenberg supports the version of WordPress.
+	if ( version_compare( $wp_version, GUTENBERG_MIN_WP_VERSION, '<' ) ) {
+		add_action( 'admin_notices', 'gutenberg_wordpress_version_too_old_notice' );
+
+		// Also add a notice to the plugin's row in the plugins list tables.
+		add_filter( 'plugin_row_meta', 'gutenberg_wordpress_version_too_old_plugin_row_meta', 10, 2 );
+
+		// Return early, do not load Gutenberg.
+		return;
```

Comment:
In the previous code, we used to disable Gutenberg plugin entirely and not just load it, should we keep disabling the plugin?

Reply:
I guess that's because we want to keep the message in the plugins list.

Reply:
Right. Also if the plugin is active but not loaded, these warning messages will be shown every time a user goes there. If the plugin is deactivated (usually on the first run after WP update) the message will only be shown once, and some users may miss it.